packages/ui/src/components/FormRenderer/components/Select/index.tsx

```diff
@@ -0,0 +1,118 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import { FC, memo } from 'react';
+import SelectComponent from '@cloudscape-design/components/select';
+import Multiselect from '@cloudscape-design/components/multiselect';
+import Autosuggest from '@cloudscape-design/components/autosuggest';
+import FormField from '@cloudscape-design/components/form-field';
+import useFieldApi, { UseFieldApiConfig } from '@data-driven-forms/react-form-renderer/use-field-api';
+import useUniqueId from '../../../../hooks/useUniqueId';
+import getErrorText from '../../utils/getErrorText';
+import { Option } from '../../types';
+
+const AUTOSUGGEST_DEFAULT_EMPTY_MESSAGE = 'No matches found';
+
+const Select: FC<UseFieldApiConfig> = (props) => {
+    const {
```

Comment:
I can see that there is a lot of repeating logic according to `@data-driven-forms`. Maybe you can use HOC approach:
```js
const withFormField = (
  Component,
  formFieldProps,
) => (props: P) => {
  const {...} = useFieldApi(props);
  return (
    <FormField
      {...}
    >
      <Component {...props} />
    </FormField>
  );
};

export default withFormField;
```

and use it:
```js
const InputWithFormField = withFormField(Input);

export default InputWithFormField;
```

I guess you understand my idea

Reply:
That is really good suggestion. Will refactor later when I have time